### PR TITLE
Bugfix/snippet bug fixes

### PR DESCRIPTION
**Pull request by lagartoverde**
Fixed the bug arne discovered about the editorDocument not resetting and also discovered the did-insert div was messing up with the scrolling so I changed that

### CODE REVIEW

File: app/controllers/snippets-management/edit/edit-snippet.js [hunk 1]
```diff
@@ -211,12 +212,13 @@ export default class SnippetsManagementEditSnippetController extends Controller
...
-  fetchEditorDocument = task(async () => {
-    this._editorDocument = await this.model.currentVersion;
+  currentVersion = trackedFunction(this, async () => {
+    const currentVersion = await this.model.currentVersion;
+    return currentVersion;
   });
 
   get editorDocument() {
-    return this._editorDocument || this.model.currentVersion;
+    return this._editorDocument || this.currentVersion.value;
```
Comment (x-m-el) [inline app/controllers/snippets-management/edit/edit-snippet.js:221]: I don't really see why `this._editorDocument` is needed. Isn't it always equal to `this.currentVersion`?

File: app/controllers/snippets-management/edit/edit-snippet.js [hunk 2]
```diff
@@ -211,12 +212,13 @@ export default class SnippetsManagementEditSnippetController extends Controller
     return this.editorDocument.content !== this.editor?.htmlContent;
   }
 
-  fetchEditorDocument = task(async () => {
-    this._editorDocument = await this.model.currentVersion;
+  currentVersion = trackedFunction(this, async () => {
+    const currentVersion = await this.model.currentVersion;
+    return currentVersion;
```
Comment (x-m-el) [inline app/controllers/snippets-management/edit/edit-snippet.js:217]: maybe
```suggestion
   return await this.model.currentVersion;
```
?

**Comment by x-m-el** [comment]
could also replace `const currentVersion = await this.model.currentVersion;` with the resource in the `save` function. (or just use `this.currentVersion` directly everywhere)

File: app/controllers/snippets-management/edit/edit-snippet.js (hunk 1 above)
Comment (lagartoverde) [reply app/controllers/snippets-management/edit/edit-snippet.js]: It was needed before, but with the current changes you are right, I didn't need it

File: app/routes/snippets-management/edit/edit-snippet.js [hunk 3]
```diff
@@ -12,4 +12,9 @@ export default class SnippetsManagementEditSnippetRoute extends Route {
     });
     return container[0];
   }
+
+  setupController(controller, model) {
+    super.setupController(controller, model);
+    controller._editorDocument = null;
+  }
```
Comment (x-m-el) [inline app/routes/snippets-management/edit/edit-snippet.js:19]: I'm guessing this isn't needed anymore either? :)

**Review by x-m-el — CHANGES_REQUESTED** [review]

File: app/routes/snippets-management/edit/edit-snippet.js (hunk 3 above)
Comment (lagartoverde) [reply app/routes/snippets-management/edit/edit-snippet.js]: Yes, you are totally right, good catch

**Review by abeforgit — APPROVED** [review]

**Pull request merged**